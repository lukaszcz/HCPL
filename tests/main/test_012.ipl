
let fact = \x { if x = 1 then 1 else x * fact (x - 1) };
let q = '(fact 10);

print (eval q);
print (reduce q);
print (reduce (reduce q));
print (eval_limited q 5);
print (eval_limited q 10);
