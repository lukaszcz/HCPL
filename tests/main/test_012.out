'3628800
'(10 * (fact 9))
'(10 * (9 * (fact 8)))
'(10 * (9 * (8 * (7 * (6 * (fact 5))))))
'3628800
